admin/projects.go

```diff
@@ -331,6 +332,32 @@ func (s *Service) triggerReconcile(ctx context.Context, depl *database.Deploymen
 
 // TriggerRefreshSource triggers refresh of a deployment's sources. If the sources slice is nil, it will refresh all sources.f
 func (s *Service) TriggerRefreshSources(ctx context.Context, depl *database.Deployment, sources []string) error {
+	// check if provided sources are exists in catalog
+	if len(sources) > 0 {
+		rt, err := s.openRuntimeClientForDeployment(depl)
+		if err != nil {
+			return s.endReconcile(ctx, depl, nil, err)
```

Comment:
Return `err`, don't call `endReconcile` in this func

Reply:
ohh, copy pst :)